mathics/core/expression.py

```diff
@@ -1103,6 +1113,9 @@ def replace_vars(self, vars, options=None,
                     leaves = [Expression('List', *func_params), body] + \
                         self.leaves[2:]
 
+        if len(vars) == 0:  # might just be a symbol set via Set[] we looked up here
```

Comment:
`benchmark.py` seems to give a very slight across the board improvement of 2-4%, but it could be as well random noise. I think it really only affects the special case of looking up a symbol that's expensive to compute.